apps/nextjs-website/src/components/organisms/Auth/__tests__/LoginForm.test.tsx

```diff
@@ -0,0 +1,139 @@
+import LoginForm from '@/components/organisms/Auth/LoginForm';
+import { fireEvent, render, act } from '@testing-library/react';
+import Wrapper from '../../../../__tests__/components/Wrapper';
+import React from 'react';
+
+describe('LoginForm', () => {
+  const promise = Promise.resolve();
+  const mockOnLogin = jest.fn(() => promise);
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should render without crashing', () => {
+    render(
+      <Wrapper>
+        <LoginForm onLogin={mockOnLogin} />
+      </Wrapper>
+    );
+  });
+
+  it('should handle username and password input', () => {
+    const { getByLabelText } = render(
+      <Wrapper>
+        <LoginForm onLogin={mockOnLogin} />
+      </Wrapper>
+    );
+
+    const usernameInput = getByLabelText(/^email$/i) as HTMLInputElement;
+    const passwordInput = getByLabelText(/^password$/i) as HTMLInputElement;
+
+    fireEvent.change(usernameInput, { target: { value: 'test@example.com' } });
+    fireEvent.change(passwordInput, { target: { value: 'password123' } });
+
+    expect(usernameInput.value).toBe('test@example.com');
+    expect(passwordInput.value).toBe('password123');
+  });
+
+  it('should call onLogin when form is submitted', async () => {
+    const { getByLabelText, getByRole } = render(
+      <Wrapper>
+        <LoginForm onLogin={mockOnLogin} />
+      </Wrapper>
+    );
+
+    const usernameInput = getByLabelText(/^email$/i) as HTMLInputElement;
+    const passwordInput = getByLabelText(/^password$/i) as HTMLInputElement;
+    const submitButton = getByRole('button', { name: /accedi/i });
+
+    fireEvent.change(usernameInput, { target: { value: 'test@example.com' } });
+    fireEvent.change(passwordInput, { target: { value: 'password123' } });
+    fireEvent.click(submitButton);
+
+    expect(mockOnLogin).toHaveBeenCalledWith({
+      username: 'test@example.com',
+      password: 'password123',
+    });
+
+    await act(() => promise);
+  });
+
+  it('should display error messages when inputs are invalid', async () => {
+    const {
+      getByLabelText,
+      findByText,
+      getByRole,
+      findAllByText,
+      queryByText,
+      queryAllByText,
+    } = render(
+      <Wrapper>
+        <LoginForm onLogin={mockOnLogin} />
+      </Wrapper>
+    );
+
+    const errorsRegex = /Questo campo non può essere vuoto/i;
```

Comment:
This test checks the value against an hardcoded string, but on the component it is a translation string.

This test will therefore break if that translation changes